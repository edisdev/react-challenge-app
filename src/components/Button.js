import React from 'react'

export default (props) => {
  let attrs = {...props}
  delete attrs.children
  attrs.className = `btn ${attrs.className||''}`
  return (<button {...attrs}>{props.children}</button>)
}