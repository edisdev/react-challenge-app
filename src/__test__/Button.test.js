import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Button from '../components/Button';


test('renders button component with props', () => {
  const { asFragment } = render(<Button id="textButton" className="btn-dark">Hello!</Button>);
  expect(asFragment()).toMatchSnapshot()
});

it('button component click event is correct work', (done) => {
  function handleClick() {
    done();
  }

  const { getByTestId } = render(
    <Button data-testid="test-button" onClick={handleClick}>Click Me!</Button>
  );

  const node = getByTestId("test-button");

  fireEvent.click(node);
})

afterEach(cleanup);