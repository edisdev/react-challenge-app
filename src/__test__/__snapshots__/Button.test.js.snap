// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button component with props 1`] = `
<DocumentFragment>
  <button
    class="btn btn-dark"
    id="textButton"
  >
    Hello!
  </button>
</DocumentFragment>
`;
